### DISCUSSION TITLE
v4.1.2

### DISCUSSION THREAD

**Discussion by ruudboon** [phalcon/cphalcon#15433] [General]
Supported PHP versions 7.3, 7.4

## Changed
- Corrected version in package.xml

<hr /><em>This discussion was created from the release <a href='https://github.com/phalcon/cphalcon/releases/tag/v4.1.2'>v4.1.2</a>.</em>

**Comment by jwang628** [comment]
Hi All,

I had the following error when building from source:
libtool: link: gcc -shared  .libs/phalcon.o   -lphp7  -mtune=native -O2   -o .libs/phalcon.dll -Wl,--enable-auto-image-base -Xlinker --out-implib -Xlinker .libs/phalcon.dll.a
/usr/lib/gcc/x86_64-pc-cygwin/11/../../../../x86_64-pc-cygwin/bin/ld: .libs/phalcon.o:phalcon.zep.c:(.text+0x1a206): undefined reference to `php_json_decode_ex'
/usr/lib/gcc/x86_64-pc-cygwin/11/../../../../x86_64-pc-cygwin/bin/ld: .libs/phalcon.o:phalcon.zep.c:(.text+0x1a24d): undefined reference to `php_json_decode_ex'
/usr/lib/gcc/x86_64-pc-cygwin/11/../../../../x86_64-pc-cygwin/bin/ld: .libs/phalcon.o:phalcon.zep.c:(.text+0x41bb9c): undefined reference to `php_json_encode'
collect2: error: ld returned 1 exit status
make: *** [Makefile:198: phalcon.la] Error 1

Please would you shed some light?  thanks a lot in advance

**Comment by jwang628** [comment]
added: -L /usr/lib 
so, new gcc command:
gcc -I. -I/home/james/phalcon_php7.3/cphalcon-4.1.2/build/php7/64bits -DPHP_ATOM_INC -I/home/james/phalcon_php7.3/cphalcon-4.1.2/build/php7/64bits/include -I/home/james/phalcon_php7.3/cphalcon-4.1.2/build/php7/64bits/main -I/home/james/phalcon_php7.3/cphalcon-4.1.2/build/php7/64bits -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -L/usr/lib -DPHALCON_RELEASE -DHAVE_CONFIG_H -mtune=native -O2 -fomit-frame-pointer -c /home/james/phalcon_php7.3/cphalcon-4.1.2/build/php7/64bits/phalcon.zep.c  -DDLL_EXPORT -DPIC -o .libs/phalcon.o

Hope that will resolve my issue.

grep php_json_ /usr/lib/json.dll.a
grep: /usr/lib/json.dll.a: binary file matches

**Comment by jwang628** [comment]
resolved.  thanks for your attention!
